ecosystem/typescript/sdk/examples/typescript/batch_transactions/batch_transactions.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  AptosAccount,
+  TxnBuilderTypes,
+  OptionalTransactionArgs,
+  AptosClient,
+  BCS,
+  MaybeHexString,
+  HexString,
+  TransactionBuilderEd25519,
+} from "aptos";
+import { Timer } from "timer-node";
+
+export type Transaction = {
+  sender: AptosAccount;
+  payload: TxnBuilderTypes.TransactionPayload;
+  extraArgs?: OptionalTransactionArgs;
+};
+
+const MAX_GAS_AMOUNT_ALLOWED = BigInt(2000000);
+
+/**
+ * This class submits banch transactions.
+ * If number of transactions is higher than the api max_batch_size config, we create `total_transactions` / `max_batch_size` transaction buffers.
+ * For each buffer we create a signed-ready-to-be-submitted transaction.
+ * To create each transaction, we fetch the current sender sequence number and maintaining a local sequence number that would be increased for every new transaction creation.
+ * We then submit each buffer to the `/transactions/batch` endpoint.
+ * We handle possible errors (for now, only `mempool_is_full` error is handled by re-submitting the failed buffer)
+ */
+export class BatchTransaction {
+  private transactions: Transaction[] = [];
+  private batchSize: number = 0;
+  private sequenceNumber: BCS.Uint64 | undefined = undefined;
+  //client = new AptosClient("http://0.0.0.0:8080/v1");
+  private client = new AptosClient("https://fullnode.devnet.aptoslabs.com");
+  private currentIndex: number = 0;
+  private currentBuffer: Transaction[] = [];
+  private latestTxnHash: string = "";
+  private lastRefreshed: Date | undefined = undefined;
+
+  private chainId: BCS.Uint8;
+  private gasUnitPrice: BCS.Uint64;
+  private maxGasAmount: BCS.Uint64;
+
+  private timer: Timer = new Timer();
+
+  async send(transactions: Transaction[], batchSize: number = 10) {
+    this.timer.start();
+    this.batchSize = batchSize;
+    this.transactions = transactions;
+    console.log(`submitting ${transactions.length} transactions`);
+    if (this.transactions.length > this.batchSize) {
+      this.currentBuffer = this.transactions.slice(this.currentIndex, this.currentIndex + this.batchSize);
+      return this.sendInBatch();
+    } else {
+      const txns = await this.createTransactions(transactions);
+      return this.client.submitBatchTransactions(txns);
+    }
+  }
+
+  async sendInBatch(): Promise<any> {
+    if (this.currentBuffer.length === 0) {
+      this.timer.stop();
+      console.log(this.timer.time());
+      return this.latestTxnHash;
+    }
+
+    const txns = await this.createTransactions(this.currentBuffer);
+
+    return this.client
+      .submitBatchTransactions(txns)
+      .then(async (data) => {
```

Comment:
I think we need better error handling here for transactions coming from the same account. Any transaction failing will cause all following transactions to be dropped